### ISSUE TITLE
Use trace messaging to indicate composite resources are being ignored

### ISSUE THREAD

**Issue by michaeltlombardi** [PowerShell/DSC#424]
### Summary of the new feature / enhancement

> As a configuration author using PSDSC resources in DSC, I always see debug messages with a warning prefix about "implementation detail not found" for composite resources when the adapter builds the DSC cache. The warning implies that something has gone wrong, but the actual behavior is desired and expected - `Invoke-DscResource` doesn't work with composite resources. Instead, the adapter should emit trace messages indicating that it's not caching the composite resource because they're not supported.

The code that is causing this behavior is in the `Invoke-DscCacheRefresh` function in the adapter module:

https://github.com/PowerShell/DSC/blob/9ee037ae9a4491d1c8a932a3224c3a281a313cc0/powershell-adapter/psDscAdapter/psDscAdapter.psm1#L78-L82

### Proposed technical implementation details (optional)

Define an additional check in the `foreach` loop to specifically discard composite resources:

```powershell
$dscResourceTypeName = $dscResource.ModuleName + '/' + $dscResource.Name

if ($dscResource.ImplementedAs -eq 'Composite') {
    $trace = @{
        'Trace' = @(
            "Skipping caching composite resource '$dscResourceTypeName'."
            "Composite resources aren't supported in DSCv3."
        ) -join ' '
    } | ConvertTo-Json -Compress
    $host.ui.WriteErrorLine($trace)
    continue
}
```

> [!NOTE]
> Out of scope for this issue, but the trace messages from the adapter all use the `Debug` level, when they should use the appropriate level for their message, and many of them lack sufficient context for a typical user to understand which resource the message is referring to and what the message indicates.

**Comment by anmenaga** [comment]
After #435 this issue happens only in Win-PS adapter;
In PS6+ PS Adapter this is fixed.
